examples/basic.php

```diff
@@ -1,21 +1,46 @@
 <?php
 
 use Amp\Http\Client\HttpClientBuilder;
+use Amp\Http\Client\Psr7\PsrAdapter;
 use Amp\Http\Client\Request;
 use Amp\Http\Client\Response;
 use Amp\Loop;
+use GuzzleHttp\Client;
+use Laminas\Diactoros\RequestFactory;
+use Laminas\Diactoros\ResponseFactory;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
 
 require __DIR__ . '/../vendor/autoload.php';
 
-Loop::run(static function () {
-    $httpClient = (new HttpClientBuilder)
-        ->build();
+Loop::run(
+    static function () {
+        $httpClient = (new HttpClientBuilder)
+            ->build();
 
-    /** @var Response $firstResponse */
-    $firstResponse = yield $httpClient->request(new Request('https://google.com/'));
-    yield $firstResponse->getBody()->buffer();
+        $psrAdapter = new PsrAdapter();
+        $psrResponseFactory = new ResponseFactory();
+        $psrRequestFactory = new RequestFactory();
 
-    /** @var Response $secondResponse */
-    $secondResponse = yield $httpClient->request(new Request('https://google.com/'));
-    yield $secondResponse->getBody()->buffer();
-});
+        $firstPsrRequest = $psrRequestFactory->createRequest('GET', 'https://google.com/');
+        /** @var Request $firstAmpRequest */
+        $firstAmpRequest = yield $psrAdapter->fromPsrRequest($firstPsrRequest);
+        // TODO: Investigate if this client bug or if Host header must be cleaned automatically
+        $firstAmpRequest->removeHeader('Host');
```

Comment:
I have a problem here. Seems that if Amp request contains `Host` header, it redirects in cycle. I don't know if it is a bug or expected behavior of `amp/http-client`, but it looks like a bug.